### PR TITLE
Gracefully handle interruptions in SaaS migration

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
    
* Capture the interruption signal (ctrl+c) and send the stop command to the data-transfer plugin
* Better handling node detection (depends on https://github.com/jfrog/jfrog-client-go/pull/611)

### CODE REVIEW

File: utils/progressbar/transferfileprogressbar.go [hunk 1]
```diff
@@ -134,6 +135,10 @@ func (t *TransferProgressMng) RemoveRepository() {
 	time.Sleep(ProgressRefreshRate)
 }
 
+func (t *TransferProgressMng) StopGracefully() {
```
Comment (RobiNino) [inline utils/progressbar/transferfileprogressbar.go:138]: Might need to hold the bar returned from this function, and abort it with drop=true in `Quit()`. See how we handle other headlines and check if it is necessary.

**Review by RobiNino — CHANGES_REQUESTED** [review]
Like!
One more thing left to handle beside the comments I left on code:
1. We need to add a stop check to `uploadByChunks`. This would probably require exposing the stop mechanism to `transferManager`

File: artifactory/commands/transferfiles/transfer.go [hunk 2]
```diff
@@ -23,6 +26,7 @@ const (
 	fileWritersChannelSize = 500000
 	retries                = 3
 	retriesWait            = 0
+	requestsNumForStop     = 5
```
[the remark this answers is not in the document]
Comment (yahavi) [reply artifactory/commands/transferfiles/transfer.go]: I replaced it with (number of running nodes) * 3.
This should be enough according to https://www.wikiwand.com/en/Coupon_collector%27s_problem

File: artifactory/commands/transferfiles/utils.go [hunk 3]
```diff
@@ -342,3 +350,52 @@ func addErrorToChannel(errorsChannelMng *ErrorsChannelMng, file FileUploadStatus
...
+
+func stopAllRunningNodes(srcUpService *srcUserPluginService, runningNodes []string) {
+	remainingNodesToStop := make(map[string]string)
+	for _, s := range runningNodes {
+		remainingNodesToStop[s] = s
+	}
+	log.Debug("Running nodes to stop:", remainingNodesToStop)
+	for i := 0; i < len(runningNodes)*3; i++ {
+		if len(remainingNodesToStop) == 0 {
+			log.Debug("All running nodes stopped successfully")
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/utils.go:390]: All running nodes stopped successfully
-->
Transfer on all Artifactory nodes stopped successfully

File: artifactory/commands/transferfiles/utils.go [hunk 4]
```diff
@@ -342,3 +350,52 @@ func addErrorToChannel(errorsChannelMng *ErrorsChannelMng, file FileUploadStatus
...
+
+func getRunningNodes(sourceRtDetails *coreConfig.ServerDetails) ([]string, error) {
+	serviceManager, err := utils.CreateServiceManager(sourceRtDetails, retries, retriesWait, false)
+	if err != nil {
+		return nil, err
+	}
+	return serviceManager.GetRunningNodes()
+}
+
+func stopAllRunningNodes(srcUpService *srcUserPluginService, runningNodes []string) {
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/utils.go:382]: stopAllRunningNodes --> stopTransferOnArtifactoryNodes

(less of a scary name :) Let's rename the test corresponding test function as well).

File: artifactory/commands/transferfiles/utils.go [hunk 5]
```diff
@@ -342,3 +350,52 @@ func addErrorToChannel(errorsChannelMng *ErrorsChannelMng, file FileUploadStatus
...
+	remainingNodesToStop := make(map[string]string)
+	for _, s := range runningNodes {
+		remainingNodesToStop[s] = s
+	}
+	log.Debug("Running nodes to stop:", remainingNodesToStop)
+	for i := 0; i < len(runningNodes)*3; i++ {
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/utils.go:388]: Let's explain here why this loop iterates over `len(runningNodes)*3`.

File: artifactory/commands/transferfiles/utils.go [hunk 6]
```diff
@@ -342,3 +350,52 @@ func addErrorToChannel(errorsChannelMng *ErrorsChannelMng, file FileUploadStatus
...
+	}
+	return serviceManager.GetRunningNodes()
+}
+
+func stopAllRunningNodes(srcUpService *srcUserPluginService, runningNodes []string) {
+	remainingNodesToStop := make(map[string]string)
+	for _, s := range runningNodes {
+		remainingNodesToStop[s] = s
+	}
+	log.Debug("Running nodes to stop:", remainingNodesToStop)
```
Comment (eyalbe4) [inline artifactory/commands/transferfiles/utils.go:387]: Running nodes to stop --> Running Artifactory nodes to stop transfer on

**Review by eyalbe4 — COMMENTED** [review]
Some minor comments.

**Pull request merged**